src/gui/src/demoOptional.js

```diff
@@ -0,0 +1,305 @@
+const TxStatus = require('../../comm/transaction.js');
+
+function demoInit() {
+    var fs = require('fs');
+    demoData =  {
+        throughput: {
+            x: [],
+            submitted: [0],
+            succeeded: [0],
+            failed: [0]
+        },
+        latency: {
+            x: [],
+            max: [0],
+            min: [0],
+            avg: [0]
+        },
+        summary: {
+            txSub: 0,
+            txSucc: 0,
+            txFail: 0,
+            round: 0,
+        },
+        report: ''
+    }
+    for(let i = 0 ; i < demoXLen ; i++) {
+        demoData.throughput.x.push(i * demoInterval);
+        demoData.latency.x.push(i * demoInterval);
+    }
+    fs.writeFileSync(demoFile,  JSON.stringify(demoData));
+}
+module.exports.init = demoInit;
+
+// TODO: need to limit the maximum length for X-Axis
+function demoRefreshX() {
+
+    var len = demoData.throughput.submitted.length;
+    while(demoData.throughput.x.length < len) {
+        if(demoData.throughput.x.length === 0) {
+            demoData.throughput.x[0] = 0;
+        }
+        else {
+            let last = demoData.throughput.x[demoData.throughput.x.length - 1];
+            demoData.throughput.x.push(last + demoInterval);
+        }
+    }
+    len = demoData.latency.max.length;
+    while(demoData.latency.x.length < len) {
+        if(demoData.latency.x.length === 0) {
+            demoData.latency.x[0] = 0;
+        }
+        else {
+            let last = demoData.latency.x[demoData.latency.x.length - 1];
+            demoData.latency.x.push(last + demoInterval);
+        }
+    }
+}
+
+function demoAddThroughput(sub, suc, fail) {
+    demoData.throughput.submitted.push(sub/demoInterval);
+    demoData.throughput.succeeded.push(suc/demoInterval);
+    demoData.throughput.failed.push(fail/demoInterval);
+    demoData.summary.txSub  = sub;
+    demoData.summary.txSucc = suc;
+    demoData.summary.txFail = fail;
+}
+
+function demoAddLatency(max, min, avg) {
+    demoData.latency.max.push(max);
+    demoData.latency.min.push(min);
+    demoData.latency.avg.push(avg);
+}
+
+function demoRefreshData(updates, label) {
+
+    for (let i = 0; i < updates.length; i++) {
+        var sub = updates[i].submitted
+        totalSubmitted += sub
+        var submitted_transactions = updates[i].committed
+    
+        for (let j =0; j < submitted_transactions.length; j++) {
+            var transactionStatus = submitted_transactions[j].status
+            var TransactionStatus = new TxStatus(transactionStatus.id, transactionStatus.status, transactionStatus.time_create, transactionStatus.time_final, 
+                transactionStatus.result, transactionStatus.verified, transactionStatus.flags, transactionStatus.error_messages);
+                
+            if (cachedEvents.get(TransactionStatus.GetID()) == undefined) {
+                cachedEvents.set(TransactionStatus.GetID(), TransactionStatus)
+                if (label == 'query') {

```

Comment:
1. I think it's not proper to use any test case specific informations in a general benchmark module to do some patches. The tester can use any label name and the semantic is unknowable to a general benchmark module.
2. Is it necessary to define a new demo file? Can we just use the old update mechanism?